src/marqo/s2_inference/s2_inference.py

```diff
@@ -519,47 +522,69 @@ def get_model_size(model_name: str, model_properties: dict) -> (int, float):
+        model_properties (dict): Model properties (e.g., dimensions, type).
+        device (str): Target device (e.g., "cuda:0", "cpu").
+        model_auth (Optional[ModelAuth]): Authorization for downloading models.
+        max_retries (int): Number of retry attempts before failing.
+        retry_delay (int): Delay (seconds) between retries.
 
     Returns:
-        Any: _description_
+        Any: Loaded model.
+
+    Raises:
+        ModelLoadError: If loading fails after all retries.
     """
     if calling_func not in ["unit_test", "_update_available_models"]:
         raise RuntimeError(f"The function `{_load_model.__name__}` should only be called by "
-                           f"`unit_test` or `_update_available_models` for threading safeness.")
-
-    print(f"loading for: model_name={model_name} and properties={model_properties}")
+                           f"`unit_test` or `_update_available_models` for threading safety.")
 
     model_type = model_properties.get("type")
     loader = _get_model_loader(model_properties.get('name', None), model_properties)
 
-    # TODO For each refactored model class, add a new elif block here and remove the if block
-    #  once we have all models refactored
-    if model_type in (ModelType.OpenCLIP, ModelType.HF_MODEL, ModelType.HF_STELLA, ModelType.LanguageBind):
-        model = loader(
-            device=device,
-            model_properties=model_properties,
-            model_auth=model_auth,
-        )
-    else:
-        model = loader(
-            model_properties.get('name', None),
-            device=device,
-            embedding_dim=model_properties['dimensions'],
-            model_properties=model_properties,
-            model_auth=model_auth,
-            max_seq_length=model_properties.get('tokens', get_default_seq_length())
-        )
-    model.load()
-    return model
+    attempt = 0
+    while attempt < max_retries:
+        try:
+            print(f"Attempt {attempt+1}/{max_retries}: Loading model `{model_name}` on `{device}`...")
+
+            # Load the model
+            if model_type in (ModelType.OpenCLIP, ModelType.HF_MODEL, ModelType.HF_STELLA, ModelType.LanguageBind):
+                model = loader(
+                    device=device,
+                    model_properties=model_properties,
+                    model_auth=model_auth,
+                )
+            else:
+                model = loader(
+                    model_properties.get('name', None),
+                    device=device,
+                    embedding_dim=model_properties['dimensions'],
+                    model_properties=model_properties,
+                    model_auth=model_auth,
+                    max_seq_length=model_properties.get('tokens', get_default_seq_length())
+                )
+
+            model.load()  # Load the model
+            print(f"✅ Model `{model_name}` loaded successfully on `{device}`.")
```

Comment:
I don't think we've ever used emojis in our output messages before 😅 but it should probably be fine